terraso_backend/apps/core/models/groups.py

```diff
@@ -40,11 +41,13 @@ class Group(SlugModel):
 
     MEMBERSHIP_TYPE_OPEN = "open"
     MEMBERSHIP_TYPE_CLOSED = "closed"
+    MEMBERSHIP_TYPE_RESTRICTED = "restricted"
```

Comment:
Question: What will happen with the current queries for groups? will the project groups return? I think maybe what is more relevant right now (if we are not able to create the separate memberhip entity) is to have a `hidden` type for the groups that should not appear, but I think this is the biggest concern on not having the separate entity as part of this implementation, since `hidden` is not that clean in terms of the meaning

Reply:
@shrouxm I don't really like our current solution, but what is happening is that the graph QL query from the client is explicilty filtering those, with the creation of a separate memberhips entity this will also work better I think. See: https://github.com/techmatters/terraso-web-client/blob/main/src/group/groupService.js#L116